### PR TITLE
Convert Boring Components to TypeScript

### PR DESCRIPTION
**Pull request by Khazakdk**
Converted a few additional related files while I was at it.  Tagging rogue specifically for this PR because I had to modify a couple files that were using the resource lookup and want to make sure the changes are sane. @darthelit

### CODE REVIEW

**Comment by tyndivelspaz** [comment]
Looks good. Doesn't seem like anything SHOULD be affected by the change and rogue should still work without having `undefined` returned.

**Review by tyndivelspaz — APPROVED** [review]

File: src/parser/rogue/outlaw/modules/talents/BladeRush.tsx [hunk 1]
```diff
@@ -21,20 +21,23 @@ class BladeRush extends Analyzer {
   }
 
   onFinishMove(event: CastEvent) {
-    if (!getResource(event.classResources, RESOURCE_TYPES.COMBO_POINTS.id)) {
+    if (!event.classResources || !getResource(event.classResources!, RESOURCE_TYPES.COMBO_POINTS.id)) {
```
Comment (MartijnHols) [inline src/parser/rogue/outlaw/modules/talents/BladeRush.tsx]:
```suggestion
    if (!event.classResources || !getResource(event.classResources, RESOURCE_TYPES.COMBO_POINTS.id)) {
```

The first check should already tell TS that event.classResources now no longer can be undefined.

**Review by MartijnHols — CHANGES_REQUESTED** [review]

File: src/parser/rogue/outlaw/modules/talents/BladeRush.tsx (hunk 1 above)
Comment (tyndivelspaz) [reply src/parser/rogue/outlaw/modules/talents/BladeRush.tsx]: Good catch. I need to get out if JS mode and into TS mode to catch things like this when reviewing.

File: src/parser/rogue/outlaw/modules/spells/BladeFlurry.ts [hunk 2]
```diff
@@ -20,14 +20,16 @@ class BladeFlurry extends Analyzer {
   }
 
   onFinishMove(event: CastEvent) {
-    if (!getResource(event.classResources, RESOURCE_TYPES.COMBO_POINTS.id)) {
+    if (!event.classResources || !getResource(event.classResources!, RESOURCE_TYPES.COMBO_POINTS.id)) {
```
Comment (Khazakdk) [inline src/parser/rogue/outlaw/modules/spells/BladeFlurry.ts]:
```suggestion
    if (!event.classResources || !getResource(event.classResources, RESOURCE_TYPES.COMBO_POINTS.id)) {
```

**Comment by Pewtro** [comment]
Looks good to me, thanks!

**Pull request merged**